packages/mui-utils/src/resolveProps.test.ts

```diff
@@ -44,4 +44,28 @@ describe('resolveProps', () => {
       foo: '',
     });
   });
+
+  it('merge components and componentsProps props', () => {
+    expect(
+      resolveProps(
+        { components: { Input: 'Input' }, componentsProps: { input: 'input' } },
```

Comment:
```suggestion
        { components: { Input: 'Input' }, componentsProps: { input: { prop: 'input' } } },
```

This is a more valid use-case, same for the `slotProps`.

Reply:
So the result would be?:

```js
{
  components: { Root: 'Root', Input: 'Input' },
  componentsProps: { root: 'root', input: { prop: 'input' } },
}
```

Reply:
Ah, I think I got your point.